subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/JavaLibraryInitIntegrationTest.groovy

```diff
@@ -206,6 +208,8 @@ class JavaLibraryInitIntegrationTest extends AbstractInitIntegrationSpec {
         ].combinations()
     }
 
+    // Spock pulls an old version of Groovy that doesn't work with Java 18
+    @IgnoreIf({ JavaVersion.current() >= JavaVersion.VERSION_18 })
```

Comment:
I think we should use a slightly different approach here. IIUC, these tests are disabled for now, and we'll re-enable them in the future. To quickly find out these tests, we can create a new `TestPrecondition` here: https://github.com/gradle/gradle/blob/c7a906e9fd84b9a4b91ed6fa467bf89cf9bd56e8/subprojects/internal-testing/src/main/groovy/org/gradle/util/TestPrecondition.groovy#L22

```
    JDK18_OR_EARLIER({
        JavaVersion.current() <= JavaVersion.VERSION_18
    }),
    JDK18_OR_EARLIER_OR_NOT_LINUX({
        JavaVersion.current() <= JavaVersion.VERSION_18 || NOT_WINDOWS.isFullFilled
    }),
```

WDYT?

Reply:
+1 this is a good idea.  I wouldn't mix the Linux thing in there though.  Just have two different preconditions that we check.

Reply:
This has been done.